tofhir-log-server/src/main/scala/io/tofhir/server/service/ExecutionService.scala

```diff
@@ -144,23 +155,45 @@ class ExecutionService() extends LazyLogging {
               StructField("startTime", StringType) ::
               StructField("errorStatus", StringType) :: Nil
           )))
-          // page size is 10, handle pagination
+          val pageSize: Int = queryParams.getOrElse("rowsPerPage", "10").toInt
+          // handle the pagination according to the page size
           val total = executionLogs.count()
```

Comment:
This variable does not make sense any more becase we have a filtering. Please modify the code so that start and end variables are calculated based on the filtered logs correctly.